mobile/Screen/MapScreen.js

```diff
@@ -119,7 +157,18 @@ function MapScreen({ navigation }) {
       mapRef.regionTimeout = setTimeout(() => {
         if (mapRef.index !== index) {
           mapRef.index = index;
-          const { coordinate } = state.markers[index];
+          const coordinate = allCards[index].location.coordinates;
+          mapRef.current.animateToRegion(
+            {
+              ...coordinate,
+              latitudeDelta: state.region.latitudeDelta,
+              longitudeDelta: state.region.longitudeDelta,
+            },
+            350,
+          );
+        }
```

Comment:
This is repeated code between the two blocks, so it's probably best to consolidate